Apps/Common/test/unit/Services/CommunicationServiceTests.cs

```diff
@@ -240,52 +238,45 @@ public void ShouldProcessChangeEvent(string action, Scenario scenario, bool cach
                     TotalResultCount = 1,
                 };
 
-                memoryCache = CreateCache(cacheEntry, cacheKey);
-            }
-            else
-            {
-                memoryCache = CreateCache();
+                this.CreateCache(cacheEntry, cacheKey);
             }
 
-            ICacheProvider cacheProvider = new MemoryCacheProvider(memoryCache);
             Mock<ICommunicationDelegate> communicationDelegateMock = new();
 
             ICommunicationService service = new CommunicationService(
                 new Mock<ILogger<CommunicationService>>().Object,
                 communicationDelegateMock.Object,
-                cacheProvider);
+                this.cacheProvider);
 
             service.ProcessChange(changeEvent);
-            RequestResult<Communication>? cacheResult = cacheProvider.GetItem<RequestResult<Communication>>(cacheKey);
+            RequestResult<Communication>? cacheResult = this.cacheProvider.GetItem<RequestResult<Communication>>(cacheKey);
             switch (scenario)
             {
                 case Scenario.Active:
                     if (!cached || (cached && !cacheMiss))
                     {
-                        Assert.Same(communication, cacheResult?.ResourcePayload);
+                        Assert.Equal(communication.Id, cacheResult!.ResourcePayload!.Id);
                     }
                     else
                     {
-                        Assert.NotEqual(communication.Id, cacheResult?.ResourcePayload?.Id);
+                        Assert.NotEqual(communication.Id, cacheResult!.ResourcePayload!.Id);
                     }
 
                     break;
 
                 case Scenario.Expired:
-                    Assert.Equal(ResultType.Success, cacheResult?.ResultStatus);
-                    Assert.Equal(0, cacheResult?.TotalResultCount);
+                    Assert.Equal(ResultType.Success, cacheResult!.ResultStatus);
```

Comment:
The null conditional is a more appropriate than null suppression as cacheResult is nullable.  This applies to additional Asserts.

Reply:
I'd agree from a consuming client point of view, but I'd argue that unit tests must be deterministic - there shouldn't be 'if/maybe/perhaps' in assertions as it would make them unreliable.

Reply:
Fair enough and if the unit test blows up things will still fail so no harm :)

Reply:
Exactly, if it blows up, then the arrange is incorrect of the logic is incorrect